answerking_app/tests/test_categories.py

```diff
@@ -253,92 +242,36 @@ def test_post_invalid_details_returns_bad_request(self):
         self.assertEqual(expected_failure_error, actual)
         self.assertEqual(response.status_code, 400)
 
-    def test_put_valid_with_items_returns_ok(self):
-        # Arrange
-        old_category = client.get(
-            f"/api/categories/{self.test_cat_1.id}"
-        ).json()
-        new_item: ItemType = {
-            "id": self.test_item_3.id,
-            "name": self.test_item_3.name,
-            "price": f"{self.test_item_3.price:.2f}",
-            "description": self.test_item_3.description,
-            "stock": self.test_item_3.stock,
-            "calories": self.test_item_3.calories,
-        }
-        post_data: NewCategoryType = {
-            "name": "New Name",
-            "items": [new_item],
-        }
-        expected_id: IDType = {"id": self.test_cat_1.id}
-        expected: CategoryType = {**post_data, **expected_id}
-
-        expected["items"] = [
-            {
-                "id": self.test_item_1.id,
-                "name": self.test_item_1.name,
-                "price": f"{self.test_item_1.price:.2f}",
-                "description": self.test_item_1.description,
-                "stock": self.test_item_1.stock,
-                "calories": self.test_item_1.calories,
-            },
-            {
-                "id": self.test_item_2.id,
-                "name": self.test_item_2.name,
-                "price": f"{self.test_item_2.price:.2f}",
-                "description": self.test_item_2.description,
-                "stock": self.test_item_2.stock,
-                "calories": self.test_item_2.calories,
-            },
-            new_item,
-        ]
-
-        # Act
-        response = client.put(
-            f"/api/categories/{self.test_cat_1.id}",
-            post_data,
-            content_type="application/json",
-        )
-        actual = response.json()
-
-        updated_category: Category = Category.objects.filter(name="New Name")[
-            0
-        ]
-        updated_list: QuerySet[Category] = Category.objects.all()
-
-        # Assert
-        self.assertNotEqual(old_category, actual)
-        self.assertIn(updated_category, updated_list)
-        self.assertEqual(expected, actual)
-        self.assertEqual(response.status_code, 200)
-
     def test_put_valid_without_items_returns_ok(self):
         # Arrange
         old_category = client.get(
             f"/api/categories/{self.test_cat_1.id}"
         ).json()
 
-        post_data: NewCategoryType = {"name": "New Name", "items": []}
```

Comment:
why was empty items removed

Reply:
Because if items is included this updated the items to an empty list which the test did not want. It is to do with the difference in the way the mixins and the old service layer worked.

Reply:
Also now using the mixins, if you want to change the items in an order you use teh put and the delete APIs from OrderItemListView

Reply:
No, it was overriding the items to an empty list when I tested it which is not what the test wants. The functionality may have changed since.

Reply:
Just tested it, it works with or without @JoeCSykes

Reply:
great